cirq-google/cirq_google/engine/simulated_local_job.py

```diff
@@ -121,14 +114,27 @@ def _execute_results(self) -> Sequence[cirq.Result]:
         Returns: a List of results from the sweep's execution.
         """
         reps, sweeps = self.get_repetitions_and_sweeps()
-        program = self.program().get_circuit()
+        parent = self.program()
+        batch_size = parent.batch_size()
         try:
             self._state = quantum.enums.ExecutionStatus.State.RUNNING
-            results = self._sampler.run_sweep(
-                program=program, params=sweeps[0] if sweeps else None, repetitions=reps
-            )
-            self._state = quantum.enums.ExecutionStatus.State.SUCCESS
-            return results
+            if batch_size == 1:
+                program = parent.get_circuit()
+                results = self._sampler.run_sweep(
+                    program=program, params=sweeps[0] if sweeps else None, repetitions=reps
+                )
+                self._state = quantum.enums.ExecutionStatus.State.SUCCESS
+                return [results]
```

Comment:
can you not just run_batch with one program? That is, why do you need these two if cases?